tests/ieee802154_hal/main.c

```diff
@@ -722,16 +693,170 @@ static int _caps_cmd(int argc, char **argv)
+
+static int send(uint8_t *dst, size_t dst_len, iolist_t iol_data, size_t num, size_t time)
+{
+    uint8_t flags;
+    uint8_t mhr[IEEE802154_MAX_HDR_LEN];
+    int mhr_len;
+
+    le_uint16_t src_pan, dst_pan;
+
+    flags = IEEE802154_FCF_TYPE_DATA;// | IEEE802154_FCF_ACK_REQ;
+
+    src_pan = byteorder_btols(byteorder_htons(CONFIG_IEEE802154_DEFAULT_PANID));
+    dst_pan = byteorder_btols(byteorder_htons(CONFIG_IEEE802154_DEFAULT_PANID));
+    uint8_t src_len = IEEE802154_LONG_ADDRESS_LEN;
+    void *src = &ext_addr;
+
+    received_acks = 0;
+    received_packets = 0;
+    send_packets = 0;
+
+    for(size_t i = 0; i < num; i++) {
+        /* fill MAC header, seq should be set by device */
+        if ((mhr_len = ieee802154_set_frame_hdr(mhr, src, src_len,
+                                            dst, dst_len,
+                                            src_pan, dst_pan,
+                                            flags, seq++)) < 0) {
+            puts("txtsnd: Error preperaring frame");
+            return 1;
+        }
+
+        iolist_t iol_hdr = {
+            .iol_next = &iol_data,
+            .iol_base = mhr,
+            .iol_len = mhr_len,
+        };
+        _send(&iol_hdr);
+        xtimer_msleep(time);
+    }
+    if (ENABLE_DEBUG) {
+        puts("-------Summary of the test-------");
+        printf("Send Packets: %d\n", send_packets);
+        printf("Acknowledged Packets: %d\n", received_acks);
+        printf("Percentage: %d\n", (received_acks * 100)/num);
+        printf("Received Packets: %d\n", received_packets);
+        puts("---------------------------------");
+    }
+    return 0;
+}
+
+int txtsnd(int argc, char **argv)
+{
+    uint8_t addr[IEEE802154_LONG_ADDRESS_LEN];
+    size_t res;
+    size_t len;
+
+    if (argc != 3) {
+        puts("Usage: txtsnd <long_addr> <len>");
+        return 1;
+    }
+    res = _parse_addr(addr, sizeof(addr), argv[1]);
+    if (res == 0) {
+        puts("Usage: txtsnd <long_addr> <len>");
+        return 1;
+    }
+    len = atoi(argv[2]);
+    iolist_t iol_data = {
+        .iol_base = payload,
+        .iol_len = len,
+        .iol_next = NULL,
+    };
+    return send(addr, res, iol_data, 1, 0);
+}
+
+iolist_t iol_data_spam(void) {
+    uint8_t *channel_ptr = &current_channel;
+    iolist_t iol_data = {
+        .iol_base = channel_ptr,
+        .iol_len = sizeof(*channel_ptr),
+        .iol_next = NULL,
+    };
+    return iol_data;
+}
+
+int txtspam(int argc, char **argv) {
+    uint8_t addr[IEEE802154_LONG_ADDRESS_LEN];
+    size_t res;
+    size_t num;
+    size_t time;
+
+    if (argc != 4) {
+        puts("Usage: txtspam <long_addr> <number of packets> <time in ms between packets>");
+        return 1;
+    }
+    res = _parse_addr(addr, sizeof(addr), argv[1]);
+    if (res == 0) {
+        puts("Usage: txtspam <long_addr> <number of packets> <time in ms between packets>");
+        return 1;
+    }
```

Comment:
Regardless of the proposal described above, you can condense these snippets:
```suggestion
    if (argc != 4 || !(res = _parse_addr(addr, sizeof(addr), argv[1]))) {
        puts("Usage: txtspam <long_addr> <number of packets> <time in ms between packets>");
        return 1;
    }
```

Reply:
I just noticed this comes from the old `txtsnd` function. You can either leave it as it is or try to condense them :)